From 763a2abf753acc409019e97f46eb3cee906c26ee Mon Sep 17 00:00:00 2001
From: Yanling Song <songyl@ramaxel.com>
Date: Tue, 9 Nov 2021 11:33:00 +0800
Subject: [PATCH] spfc: Fix compile errors when O=xxx is specified
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 763a2abf753acc409019e97f46eb3cee906c26ee

Ramaxel inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4DBD7
CVE: NA

-----------------------------------------

Fix two compile errors:
1. Compile failed when O=xxx is specified;
2. Compile conflicted when spfc and spnic are compiled in parallel
by -j option because they share some .c files.

make O=tmp -j200  > build.log

In file included from ../drivers/scsi/spfc/hw/spfc_utils.c:4:0:
../drivers/scsi/spfc/hw/spfc_utils.h:7:10: fatal error: unf_type.h: No such file or directory
 #include "unf_type.h"
          ^~~~~~~~~~~~
compilation terminated.
make[4]: *** [drivers/scsi/spfc/hw/spfc_utils.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [drivers/scsi/spfc] Error 2
make[3]: *** Waiting for unfinished jobs....
../drivers/net/ethernet/ramaxel/spnic/spnic_sriov.c:9:10: fatal error: sphw_common.h: No such file or directory
 #include "sphw_common.h"
          ^~~~~~~~~~~~~~~
compilation terminated.

Signed-off-by: Yanling Song <songyl@ramaxel.com>
Reviewed-by: Zhang Lei<zhanglei48@huawei.com>
[zzk: Adjusted commit message]
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/ramaxel/spnic/Makefile |  2 +-
 drivers/scsi/spfc/Makefile                  | 30 ++++++++++-----------
 drivers/scsi/spfc/sphw_api_cmd.c            |  1 +
 drivers/scsi/spfc/sphw_cmdq.c               |  1 +
 drivers/scsi/spfc/sphw_common.c             |  1 +
 drivers/scsi/spfc/sphw_eqs.c                |  1 +
 drivers/scsi/spfc/sphw_hw_cfg.c             |  1 +
 drivers/scsi/spfc/sphw_hw_comm.c            |  1 +
 drivers/scsi/spfc/sphw_hwdev.c              |  1 +
 drivers/scsi/spfc/sphw_hwif.c               |  1 +
 drivers/scsi/spfc/sphw_mbox.c               |  1 +
 drivers/scsi/spfc/sphw_mgmt.c               |  1 +
 drivers/scsi/spfc/sphw_prof_adap.c          |  1 +
 drivers/scsi/spfc/sphw_wq.c                 |  1 +
 14 files changed, 28 insertions(+), 16 deletions(-)
 create mode 120000 drivers/scsi/spfc/sphw_api_cmd.c
 create mode 120000 drivers/scsi/spfc/sphw_cmdq.c
 create mode 120000 drivers/scsi/spfc/sphw_common.c
 create mode 120000 drivers/scsi/spfc/sphw_eqs.c
 create mode 120000 drivers/scsi/spfc/sphw_hw_cfg.c
 create mode 120000 drivers/scsi/spfc/sphw_hw_comm.c
 create mode 120000 drivers/scsi/spfc/sphw_hwdev.c
 create mode 120000 drivers/scsi/spfc/sphw_hwif.c
 create mode 120000 drivers/scsi/spfc/sphw_mbox.c
 create mode 120000 drivers/scsi/spfc/sphw_mgmt.c
 create mode 120000 drivers/scsi/spfc/sphw_prof_adap.c
 create mode 120000 drivers/scsi/spfc/sphw_wq.c

diff --git a/drivers/net/ethernet/ramaxel/spnic/Makefile b/drivers/net/ethernet/ramaxel/spnic/Makefile
index f86ccff374f6..207e1d9c431a 100644
--- a/drivers/net/ethernet/ramaxel/spnic/Makefile
+++ b/drivers/net/ethernet/ramaxel/spnic/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_SPNIC) += spnic.o
 
-subdir-ccflags-y += -I$(src)/hw
+subdir-ccflags-y += -I$(srctree)/$(src)/hw
 
 spnic-objs := hw/sphw_common.o \
 		hw/sphw_hwif.o \
diff --git a/drivers/scsi/spfc/Makefile b/drivers/scsi/spfc/Makefile
index 02fe0213e048..849b730ac733 100644
--- a/drivers/scsi/spfc/Makefile
+++ b/drivers/scsi/spfc/Makefile
@@ -1,9 +1,9 @@
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_SPFC) += spfc.o
 
-subdir-ccflags-y += -I$(src)/../../net/ethernet/ramaxel/spnic/hw
-subdir-ccflags-y += -I$(src)/hw
-subdir-ccflags-y += -I$(src)/common
+subdir-ccflags-y += -I$(srctree)/$(src)/../../net/ethernet/ramaxel/spnic/hw
+subdir-ccflags-y += -I$(srctree)/$(src)/hw
+subdir-ccflags-y += -I$(srctree)/$(src)/common
 
 spfc-objs := common/unf_init.o \
 		common/unf_event.o \
@@ -33,15 +33,15 @@ spfc-objs := common/unf_init.o \
 		hw/spfc_cqm_bitmap_table.o \
 		hw/spfc_cqm_main.o \
 		hw/spfc_cqm_object.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_hwdev.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_hw_cfg.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_hw_comm.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_prof_adap.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_common.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_hwif.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_wq.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_cmdq.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_eqs.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_mbox.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_mgmt.o \
-		../../net/ethernet/ramaxel/spnic/hw/sphw_api_cmd.o
+		sphw_hwdev.o \
+		sphw_hw_cfg.o \
+		sphw_hw_comm.o \
+		sphw_prof_adap.o \
+		sphw_common.o \
+		sphw_hwif.o \
+		sphw_wq.o \
+		sphw_cmdq.o \
+		sphw_eqs.o \
+		sphw_mbox.o \
+		sphw_mgmt.o \
+		sphw_api_cmd.o
diff --git a/drivers/scsi/spfc/sphw_api_cmd.c b/drivers/scsi/spfc/sphw_api_cmd.c
new file mode 120000
index 000000000000..27c7c0770fa3
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_api_cmd.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_api_cmd.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_cmdq.c b/drivers/scsi/spfc/sphw_cmdq.c
new file mode 120000
index 000000000000..5ac779ba274b
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_cmdq.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_cmdq.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_common.c b/drivers/scsi/spfc/sphw_common.c
new file mode 120000
index 000000000000..a1a30a4840e1
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_common.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_common.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_eqs.c b/drivers/scsi/spfc/sphw_eqs.c
new file mode 120000
index 000000000000..74430dcb9dc5
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_eqs.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_eqs.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_hw_cfg.c b/drivers/scsi/spfc/sphw_hw_cfg.c
new file mode 120000
index 000000000000..4f43d68624c1
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_hw_cfg.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_hw_cfg.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_hw_comm.c b/drivers/scsi/spfc/sphw_hw_comm.c
new file mode 120000
index 000000000000..c943b3b2933a
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_hw_comm.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_hw_comm.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_hwdev.c b/drivers/scsi/spfc/sphw_hwdev.c
new file mode 120000
index 000000000000..b7279f17eaa2
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_hwdev.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_hwdev.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_hwif.c b/drivers/scsi/spfc/sphw_hwif.c
new file mode 120000
index 000000000000..d40ef71f9033
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_hwif.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_hwif.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_mbox.c b/drivers/scsi/spfc/sphw_mbox.c
new file mode 120000
index 000000000000..1b00fe7289cc
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_mbox.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_mbox.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_mgmt.c b/drivers/scsi/spfc/sphw_mgmt.c
new file mode 120000
index 000000000000..fd18a73e9d3a
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_mgmt.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_mgmt.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_prof_adap.c b/drivers/scsi/spfc/sphw_prof_adap.c
new file mode 120000
index 000000000000..fbc7db05dd27
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_prof_adap.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_prof_adap.c
\ No newline at end of file
diff --git a/drivers/scsi/spfc/sphw_wq.c b/drivers/scsi/spfc/sphw_wq.c
new file mode 120000
index 000000000000..cdfcb3a610c0
--- /dev/null
+++ b/drivers/scsi/spfc/sphw_wq.c
@@ -0,0 +1 @@
+../../net/ethernet/ramaxel/spnic/hw/sphw_wq.c
\ No newline at end of file
-- 
2.26.2

